theano/tensor/elemwise.py

```diff
@@ -502,6 +503,7 @@ class Elemwise(OpenMPOp):
     | Elemwise(log)(rand(3, 4, 5))
 
     """
+    __props__ = ("scalar_op", "inplace_pattern", "name", "nfunc_spec", "openmp")
```

Comment:
The better solution would be, Except those classes from the condition to have `__props__` and not cache them also ?